src/workerd/api/web-socket.c++

```diff
@@ -10,6 +10,35 @@
 
 namespace workerd::api {
 
+WebSocket::WebSocket(kj::WebSocket& ws)
+    : url(nullptr),
+      farNative(nullptr),
+      outgoingMessages(IoContext::current().addObject(kj::heap<OutgoingMessagesMap>())),
+      locality(LOCAL) {
+  // This constructor is used when reinstantiating a websocket that had been hibernating, which is
+  // why we can go straight to the Accepted state. However, note that we are actually in the
+  // `Hibernatable` "sub-state" as a consequence of passing a kj::WebSocket& to the constructor!
+  auto nativeObj = kj::heap<Native>();
+  nativeObj->state.init<Accepted>(
+      Accepted::Hibernatable{.ws = ws}, *farNative, IoContext::current());
+  farNative = IoContext::current().addObject(kj::mv(nativeObj));
+}
+
+jsg::Ref<WebSocket> WebSocket::unhibernate(
+    jsg::Lock& js,
+    kj::WebSocket& ws,
+    v8::Local<v8::Value>  attachment,
+    kj::Maybe<kj::StringPtr> url,
+    kj::Maybe<kj::StringPtr> protocol,
+    kj::Maybe<kj::StringPtr> extensions) {
+  auto websocket = jsg::alloc<WebSocket>(ws);
+  websocket->url = url.map([](kj::StringPtr s) { return kj::str(s); });
+  websocket->protocol = protocol.map([](kj::StringPtr s) { return kj::str(s); });
+  websocket->extensions = extensions.map([](kj::StringPtr s) { return kj::str(s); });
```

Comment:
Not sure if this is a better name, but how about `maybeDeepCopy` or `MaybeCopy` since you return a `kj::Maybe`? Typically when we use `kj::Maybe` we name it `maybeValue`...

Reply:
I don't use this function anymore (we `kj::mv` instead) so I'm going to remove this file. Feels wrong to add a utility that isn't used on the off chance someone might find a use for it in the future.

Any objections?

Reply:
No objections from me.